elyra/pipeline/handlers.py

```diff
@@ -137,15 +139,29 @@ class PipelineComponentHandler(HttpErrorMixin, APIHandler):
     """Handler to expose method calls to retrieve pipelines editor component configuration"""
 
     @web.authenticated
-    async def get(self, runtime):
-        self.log.debug(f'Retrieving pipeline components for: {runtime} runtime')
+    async def get(self, runtime_type):
+        self.log.debug(f"Retrieving pipeline components for runtime type: {runtime_type}")
 
         processor_manager = PipelineProcessorManager.instance()
-        if not processor_manager.is_supported_runtime_platform(runtime) and \
-                not processor_manager.is_supported_runtime(runtime):
-            raise web.HTTPError(400, f"Invalid processor name '{runtime}'")
+        if processor_manager.is_supported_runtime_type(runtime_type):
+            # The request path uses the appropriate RuntimeProcessorType name.
+            # Include generic components for all runtime types
+            components: List[Component] = ComponentCache.get_generic_components()
+
+            # Add additional runtime-type-specific components, if present
+            runtime_processor_type = RuntimeProcessorType.get_instance_by_name(runtime_type)
+            components.extend(ComponentCache.instance().get_all_components(platform=runtime_processor_type))
+        elif processor_manager.is_supported_runtime(runtime_type):
+            # The endpoint path contains the shorthand version of a runtime (e.g., 'kfp',
+            # 'airflow'). This case and its associated functions should eventually be removed
+            # in favor of using the RuntimeProcessorType name in the request path.
+            self.log.warning(f"Deprecation warning: when calling endpoint '{self.request.path}' "
+                             f"use runtime type name (e.g. 'KUBEFLOW_PIPELINES', 'APACHE_AIRFLOW') "
+                             f"instead of shorthand name (e.g., 'kfp', 'airflow')")
+            components: List[Component] = await processor_manager.get_components(runtime_type)
+        else:
+            raise web.HTTPError(400, f"Invalid runtime type '{runtime_type}'")
```

Comment:
This might be cleaner if we check for the runtime first and get its runtime type, then fall into the cache access block.  Perhaps something like this...
```python
        if processor_manager.is_supported_runtime(runtime_type):
            # The endpoint path contains the shorthand version of a runtime (e.g., 'kfp',
            # 'airflow'). This case and its associated functions should eventually be removed
            # in favor of using the RuntimeProcessorType name in the request path.
            self.log.warning(f"Deprecation warning: when calling endpoint '{self.request.path}' "
                             f"use runtime type name (e.g. 'KUBEFLOW_PIPELINES', 'APACHE_AIRFLOW') "
                             f"instead of shorthand name (e.g., 'kfp', 'airflow')")
            runtime_processor_type = processor_manager.get_runtime_type(runtime_type)
        elif processor_manager.is_supported_runtime_type(runtime_type):
            runtime_processor_type = RuntimeProcessorType.get_instance_by_name(runtime_type)
        else:
            raise web.HTTPError(400, f"Invalid runtime type '{runtime_type}'")

        # The request path uses the appropriate RuntimeProcessorType name.
        # Include generic components for all runtime types
        components: List[Component] = ComponentCache.get_generic_components()

        # Add additional runtime-type-specific components, if present
        components.extend(ComponentCache.instance().get_all_components(platform=runtime_processor_type))
```

Reply:
That's much nicer 😄